### PR TITLE
refactor: replace downshift with headless UI

### LINKED ISSUE #106: Replace downshift with headless-ui
The input width has some kind of glitch occasionally. 

image

### PR DESCRIPTION
**Pull request by Pertempto**
## What has changed

* Replaced `downshift` usage with `@headlessui/react`
* Replaced `AutocompleteInput` with `Autocomplete`

## Connected Issues

Closes #106

## QA Steps

1. Sign in with a translator account
2. Click the down icon for any word
    * A list of options should appear
3. Select an option
    * The new gloss should be saved
4. Type in the autocomplete input
    * The list of options should appear again, filtered by the input value.
5. Type a brand new gloss, then press tab
    * The new gloss should be saved

## Post-Deployment

- [ ] Nothing required

### CODE REVIEW

**Comment by Pertempto** [comment]
I included the commits from `68-language-users`, so this is targeting that until #113 is merged.  
I didn't really write any new code. I was able to bring over pieces from the multi select and the old autocomplete components, and that basically covered everything.

**Comment by Pertempto** [comment]
I forgot. This also include normalization code, so it should also cover #84. Test with unicode sequences you gave

> A letter can have multiple diacritics (ie in Hebrew שָּׁ is formed from the character \u05e9 plus these three diacritics in any order \u05c1, \u05bc, \u05b8). This means there are 6 ways of encoding that letter with those diacritics, but they render the same to the user. When a user types in a gloss, we need to connect to an existing matching gloss even if the diacritics are in a different order. Otherwise the number of ways a word has been glossed will be artificially inflated with what appears to be duplicates.

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   "dependencies": {
     "@casl/ability": "6.5.0",
     "@casl/prisma": "1.4.0",
+    "@headlessui/tailwindcss": "^0.1.3",
```
Comment (arrocke) [inline package.json]: Let's lock this version down

File: packages/web/src/shared/components/form/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  onBlur?(): void;
+  onChange?(value: string): void;
+  onCreate?(text?: string): void;
+  items: AutocompleteItem[];
+  defaultValue?: string[];
+  name: string;
+  hasErrors?: boolean;
+}
+
+const Autocomplete = ({
```
Comment (arrocke) [inline packages/web/src/shared/components/form/Autocomplete.tsx]: Can we call this AutocompleteInput to be consistent with the other input components?

**Review by arrocke — APPROVED** [review]
Autocomplete looks good! I added a few comments to #84 with a few more ideas about diacritics. Up you to you if you want to include that here or in a separate PR.

File: package.json (hunk 1 above)
Comment (Pertempto) [reply package.json]: Oops, I forgot

File: packages/web/src/shared/components/form/Autocomplete.tsx (hunk 2 above)
Comment (Pertempto) [reply packages/web/src/shared/components/form/Autocomplete.tsx]: Sure thing

**Pull request merged**